File: fstp-core-bo/resource/com/purefun/fstp/core/bo/otw/QNSRequestBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.QNSRequestBO;
import com.purefun.fstp.core.bo.pro.QNSRequestBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class QNSRequestBO_OTW implements ICommom_OTW {
    QNSRequestBO_PRO.QNSRequestBO.Builder builder = null;
    QNSRequestBO bo = null;

    public QNSRequestBO_OTW() {
        builder = QNSRequestBO_PRO.QNSRequestBO.newBuilder();
        bo= new QNSRequestBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public QNSRequestBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = QNSRequestBO_PRO.QNSRequestBO.newBuilder();
        bo= new QNSRequestBO();
        QNSRequestBO_PRO.QNSRequestBO receive = QNSRequestBO_PRO.QNSRequestBO.parseFrom(message);
        setServername(receive.getServername());
        setRequest(receive.getRequest());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public QNSRequestBO_OTW(QNSRequestBO bofrom){
        builder = QNSRequestBO_PRO.QNSRequestBO.newBuilder();
        bo= new QNSRequestBO();
        setServername(bofrom.servername);
        setRequest(bofrom.request);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public QNSRequestBO getBo() { 
        return bo;
    }

    public java.lang.String getServername() {
        return builder.getServername();
    }

    public void setServername(java.lang.String servername) {
        bo.servername = servername;
        builder.setServername(servername);
    }

    public java.lang.String getRequest() {
        return builder.getRequest();
    }

    public void setRequest(java.lang.String request) {
        bo.request = request;
        builder.setRequest(request);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "QNSRequestBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "servername = " + getServername() +"," +
            "request = " + getRequest() +"," +
         "]";
    }
}

File: fstp-python/com/purefun/fstp/core/bo/otw/QNSRequestBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.core.bo.pro import QNSRequestBO_pb2
from com.purefun.fstp.core.bo.model.QNSRequestBO import QNSRequestBO

class QNSRequestBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = QNSRequestBO_pb2.QNSRequestBO()
        self._bo = QNSRequestBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.servername = self._bo.servername
        self._bo_pro.request = self._bo.request
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.servername = self._bo_pro.servername
        self._bo.request = self._bo_pro.request
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getServername(self):
        return self._bo.servername

    def setServername(self, servername):
        self._bo.servername = servername
        self._bo_pro.servername = servername

    def getRequest(self):
        return self._bo.request

    def setRequest(self, request):
        self._bo.request = request
        self._bo_pro.request = request

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "QNSRequestBO_OTW ["+"servername = " + self.getServername() +"," +"request = " + self.getRequest() +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"
